core/arch/arm/plat-imx/psci.c

```diff
@@ -67,3 +71,67 @@ int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 
 	return PSCI_RET_SUCCESS;
 }
+
+int psci_cpu_off(void)
+{
+	uint32_t core_id;
+	vaddr_t va = core_mmu_get_va(SRC_BASE, MEM_AREA_IO_SEC);
+
+	core_id = get_core_pos();
+
+	write32(~0, va + SRC_GPR1 + core_id * 8 + 4);
+
+	DMSG("core_id: %d\n", core_id);
+
+	psci_armv7_cpu_off();
+
+	imx_set_src_gpr(core_id, ~0);
+
+	thread_mask_exceptions(THREAD_EXCP_ALL);
+
+	asm volatile(
```

Comment:
@jenswi-linaro if use "while (true)", compiler will complain errors, because this function expects a value returned.